core/src/main/java/hudson/cli/InstallToolCommand.java

```diff
@@ -62,18 +62,18 @@ public String getShortDescription() {
     }
 
     protected int run() throws Exception {
-        Jenkins h = Jenkins.getInstance();
+        Jenkins h = Jenkins.getActiveInstance();
         h.checkPermission(Jenkins.READ);
 
         // where is this build running?
         BuildIDs id = checkChannel().call(new BuildIDs());
 
         if (!id.isComplete())
-            throw new AbortException("This command can be only invoked from a build executing inside Hudson");
+            throw new IllegalStateException("This command can be only invoked from a build executing inside Hudson");
 
-        AbstractProject p = Jenkins.getInstance().getItemByFullName(id.job, AbstractProject.class);
+        AbstractProject p = h.getItemByFullName(id.job, AbstractProject.class);
         if (p==null)
-            throw new AbortException("No such job found: "+id.job);
+            throw new IllegalStateException("No such job found: "+id.job);
```

Comment:
Hmm, I did that mistake first too. The point is that neither (value of) id.job or id itself is a cmdline argument (see annotation), in that case it makes sense for me to raise IllegalStateEx instead of IllegalArgumentEx - non existing job can be taken as state as well.

IllegalArgumentEx should be in my opinion closely tied to input parameters.